src/core/lib/surface/channel.cc

```diff
@@ -251,7 +259,10 @@ grpc_channel* grpc_channel_create(const char* target,
   // accommodate that, we call grpc_init() here and then call
   // grpc_shutdown() when the channel is actually destroyed, thus
   // ensuring that shutdown is deferred until that point.
-  grpc_init();
+  if (!grpc_channel_args_find_bool(
```

Comment:
Instead of calling `grpc_channel_args_find_bool()` multiple times, please call it once and save the result in a local bool, which can be checked twice.